scheduler/src/cook/mesos/mesos_mock.clj

```diff
@@ -209,19 +211,37 @@
 (defn complete-task!
   "Marks the task as complete in the state and updates the resources
    Returns the updated state"
-  [state task-id scheduler driver task-state]
-  (log/debug "Completing task" {:task-id task-id :state task-state :task (get-in state [:task-id->task task-id])})
-  (if-let [task (get-in state [:task-id->task task-id])]
-    (do
-      (.statusUpdate scheduler driver (mesos-type/->pb :TaskStatus
-                                                       {:task-id {:value task-id}
-                                                        :state task-state}))
-      (-> state
-          (update :task-id->task #(dissoc % task-id))
-          (update :task-id->completed-task #(assoc % task-id (assoc task :complete-time (t/now))))
-          (update-in [:slave-id->host (:value (:slave-id task)) :available-resources]
-                     #(combine-resources (:resources task) %))))
-    state))
+  ([state task-id scheduler driver]
+   (complete-task! state task-id scheduler driver nil))
+  ([state task-id scheduler driver task-state]
+   (log/debug "Completing task" {:task-id task-id :state task-state :task (get-in state [:task-id->task task-id])})
+   (if-let [task (get-in state [:task-id->task task-id])]
+     (do
+       (let [task->complete-status (get-in state [:config :task->complete-status]) 
+             task-state (or task-state (task->complete-status task))]
+         (.statusUpdate scheduler driver (mesos-type/->pb :TaskStatus
+                                                          {:task-id {:value task-id}
+                                                           :state task-state}))
+         (-> state
+             (update :task-id->task #(dissoc % task-id))
+             (update :task-id->completed-task #(assoc % task-id (assoc task :complete-time (t/now))))
+             (update-in [:slave-id->host (:value (:slave-id task)) :available-resources]
+                        #(combine-resources (:resources task) %)))))
+     state)))
+
+(defn complete-tasks!
+  "Removes tasks from state that have completed and calls complete-task!"
+  [state scheduler driver]
+  (let [now (:now state)
+        time-task-id-pairs (:time-task-id-pairs state)
+        complete? (comp (partial t/after? now) first)
+        to-complete-pairs (filter complete? time-task-id-pairs)
+        remaining (remove complete? time-task-id-pairs)
```

Comment:
Ideally, yes, but the filter for the kill would be a bit time consuming. I think as is it is fine since eventually the tasks time will be up and when we go to complete it the task id won't be there and it will be a noop. 

I propose leaving it for now since the only use case for the mesos mock, the simulator, doesn't need it and the implementation is correct (though a little wasteful) as is.